### PR TITLE
Refactors before torch-ifying

### PR DESCRIPTION
**Pull request by turian**
This combines #33 #35 #36 #38 #42

### CODE REVIEW

**Comment by maxsolomonhenry** [comment]
I'm happy with this contingent on our other discussions.

File: examples/examples.py [hunk 1]
```diff
@@ -182,35 +183,46 @@ def stft_plot(signal, sample_rate=SAMPLE_RATE):
...
 # -
 
-drum_out = my_drum()
+drum_out = my_drum.npyforward()
 stft_plot(drum_out)
 ipd.Audio(drum_out, rate=vca.sample_rate)
 
+# # Random synths
+#
+# Let's generate some random synths
```
Comment (jorshi) [inline examples/examples.py:217]: This is great. Love the randomize feature.

**Review by jorshi — APPROVED** [review]
I don't love the name npyforward -- but I think it's fine for now as we are getting into the torch-fication! Looks good.

**Comment by turian** [comment]
npyforward is awful. You can rename it if you like, just avoid __call__ and forward ?

**Pull request merged**